src/main/webapp/filesFragment.xhtml

```diff
@@ -117,7 +117,7 @@
                     </div>
                 </div>
                 <div class="col-xs-7 text-right">
-                    <p:outputPanel id="filesButtons">
+                    <p:outputPanel id="completionButtons">
```

Comment:
Isn't this a duplicate ID of the new completionButtons outputPanel added to the dataset?

Reply:
~yes, but the editFileFragment is either embedded in the dataset.xhtml or filesFragement.xhtml, but not both, so I think there is only ever one element with this id. Having them the same is really the point - as the AllUploadsFinished command tries to update 'completionButtons' and fails if it can't find it, which is what happened when I tried this without having the empty element in dataset.xhtml with this name. If there's  a way to update an element only if it exists, that should work instead.